packages/amplify-cli-core/src/amplify-node-pkg-detector/parser-factory.ts

```diff
@@ -12,17 +13,20 @@ export class LockFileParserFactory {
   /**
    * get lock file Parser
    */
-  public static getLockFileParser(lockfileType: string): LockfileParser {
-    switch (lockfileType) {
+  public static getLockFileParser(packageManager: PackageManager): LockfileParser {
+    switch (packageManager.packageManager) {
       case LockfileType.NPM:
         return new PackageLockParser();
-      case LockfileType.YARN:
-        return new YarnLockParser();
-      case LockfileType.YARN2:
-        return new Yarn2LockParser();
+      case LockfileType.YARN: {
+        if (packageManager?.version?.major && packageManager?.version?.major >= 2) {
```

Comment:
should this throw if yarn version is undef ?

Reply:
I can do that, how about if I do this work as a followup taks when implementing/validating support lambda functions yarn 2 (there is a separate asana task for this)?